### PR TITLE
Fix and simplify resolveToFullPath_

### PR DESCRIPTION
**Pull request by ussuri**
Bug fixes:

* resolveToFullPath_('/', '.foo') would return '/foo' - not '/.foo'
* resolveToFullPath_('/', 'a/b/../../foo') would return '/a/foo' - not '/foo'
* resolveToFullPath_('/', 'a////b////foo') would return '/a//b//foo' - not '/a/b/foo'
* Some others.

The jsbin for the new version is here: http://jsbin.com/fiqogurowo/2/edit?js,console

### CODE REVIEW

**Comment by ebidel** [comment]
- `resolveToFullPath_('/', '.foo')` _should_ resolve to `/.foo`. ".foo" is a valid dot file name, not a folder
- `resolveToFullPath_('/', 'a/b/../../foo') _should_ return "/foo"
- In unix, it resolves as the latter.

**Comment by ussuri** [comment]
Sorry, I structured my comments in an unclear manner. Rephrasing:

* resolveToFullPath_('/', '.foo') would return '/foo' before - now returns '/.foo'
* resolveToFullPath_('/', 'a/b/../../foo') would return '/a/foo' before - now returns '/foo'
* resolveToFullPath_('/', 'a////b////foo') would return '/a//b//foo' before - now returns  '/a/b/foo'

So, it seems you are in agreement with what this PR achieves :)

**Comment by ussuri** [comment]
I'll add a couple of comments in the code to explain why some of the above was happening.

File: src/idb.filesystem.js [hunk 1]
```diff
@@ -109,46 +109,31 @@ function resolveToFullPath_(cwdFullPath, path) {
...
-  } 
-
-  // Replace './' by current dir. ('./one/./two' -> one/two)
-  fullPath = fullPath.replace(/\.\//g, DIR_SEPARATOR);
-
-  // Replace '//' with '/'.
-  fullPath = fullPath.replace(/\/\//g, DIR_SEPARATOR);
 
-  // Replace '/.' with '/'.
-  fullPath = fullPath.replace(/\/\./g, DIR_SEPARATOR);
```
Comment (ussuri) [inline src/idb.filesystem.js]: This was turning '/.foo' into '/foo'.

File: src/idb.filesystem.js [hunk 2]
```diff
@@ -109,46 +109,31 @@ function resolveToFullPath_(cwdFullPath, path) {
...
+    fullPath = cwdFullPath + DIR_SEPARATOR + path;
   }
 
-  // Adjust '..'s by removing parent directories when '..' flows in path.
+  // Normalize '.'s,  '..'s and '//'s.
   var parts = fullPath.split(DIR_SEPARATOR);
+  var finalParts = [];
   for (var i = 0; i < parts.length; ++i) {
     var part = parts[i];
-    if (part == '..') {
```
Comment (ussuri) [inline src/idb.filesystem.js]: This block could walk back one level at most.

File: src/idb.filesystem.js [hunk 3]
```diff
@@ -109,46 +109,31 @@ function resolveToFullPath_(cwdFullPath, path) {
...
-  // Add back in leading slash.
-  if (fullPath[0] != DIR_SEPARATOR) {
-    fullPath = DIR_SEPARATOR + fullPath;
-  } 
-
-  // Replace './' by current dir. ('./one/./two' -> one/two)
-  fullPath = fullPath.replace(/\.\//g, DIR_SEPARATOR);
-
-  // Replace '//' with '/'.
-  fullPath = fullPath.replace(/\/\//g, DIR_SEPARATOR);
```
Comment (ussuri) [inline src/idb.filesystem.js]: This would perform only one round of `//` -> `/` conversion, so `////` would only convert to `//`, but not further.

**Comment by ussuri** [comment]
@ebidel: ping

File: src/idb.filesystem.js [hunk 4]
```diff
@@ -109,46 +109,31 @@ function resolveToFullPath_(cwdFullPath, path) {
...
   var parts = fullPath.split(DIR_SEPARATOR);
+  var finalParts = [];
   for (var i = 0; i < parts.length; ++i) {
     var part = parts[i];
-    if (part == '..') {
-      parts[i - 1] = '';
-      parts[i] = '';
+    if (part === '..') {
+      // Go up one level.
+      if (!finalParts.length)
```
Comment (ebidel) [inline src/idb.filesystem.js]: use `{}` please

**Comment by ebidel** [comment]
LGTM. Tests are passing :) Can you just address the one comment.

File: src/idb.filesystem.js (hunk 4 above)
Comment (ussuri) [inline src/idb.filesystem.js]: done

**Comment by ussuri** [comment]
Done!

**Pull request merged**